src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java

```diff
@@ -0,0 +1,348 @@
+     *                    according to standard coordinate-sorted BAM conventions
+     * @param downsamplingMethod information about how to downsample the reads
+     * @param includeReadsWithDeletionAtLoci Include reads with deletion at loci
+     * @param keepUniqueReadListInLIBS Keep unique read list in LIBS
+     * @param genomeLocParser used to create genome locs
+     * @param samples a complete list of samples present in the read groups for the reads coming from samIterator.
+     *                This is generally just the set of read group sample fields in the SAMFileHeader.  This
+     *                list of samples may contain a null element, and all reads without read groups will
+     *                be mapped to this null sample
+     */
+    public LocusIteratorByState(final Iterator<GATKRead> samIterator,
+                                final DownsamplingMethod downsamplingMethod,
+                                final boolean includeReadsWithDeletionAtLoci,
+                                final boolean keepUniqueReadListInLIBS,
+                                final GenomeLocParser genomeLocParser,
+                                final Collection<String> samples,
+                                final SAMFileHeader header) {
+        this(samIterator,
+                DownsamplingMethod.toDownsamplingInfo(downsamplingMethod),
+                includeReadsWithDeletionAtLoci,
+                genomeLocParser,
+                samples,
+                keepUniqueReadListInLIBS,
+                header);
+    }
+
+    /**
+     * Create a new LocusIteratorByState
+     *
+     * @param samIterator the iterator of reads to process into pileups.  Reads must be ordered
+     *                    according to standard coordinate-sorted BAM conventions
+     * @param downsamplingInfo meta-information about how to downsampling the reads
+     * @param genomeLocParser used to create genome locs
+     * @param samples a complete list of samples present in the read groups for the reads coming from samIterator.
+     *                This is generally just the set of read group sample fields in the SAMFileHeader.  This
+     *                list of samples may contain a null element, and all reads without read groups will
+     *                be mapped to this null sample
+     * @param maintainUniqueReadsList if true, we will keep the unique reads from off the samIterator and make them
+     *                                available via the transferReadsFromAllPreviousPileups interface
+     */
+    public LocusIteratorByState(final Iterator<GATKRead> samIterator,
+                                final LIBSDownsamplingInfo downsamplingInfo,
+                                final boolean includeReadsWithDeletionAtLoci,
+                                final GenomeLocParser genomeLocParser,
+                                final Collection<String> samples,
+                                final boolean maintainUniqueReadsList,
```

Comment:
Please make sure that we are always setting `maintainUniqueReadsList` to `false` everywhere we are constructing a `LocusIteratorByState` in non-test code -- this is a feature we will probably not use in GATK4, and will likely eventually want to rip out.